### PR TITLE
Fixes example in Type.asSeenFrom

### PR DESCRIPTION
**Pull request by densh**
It was written in some form of non-executable pseudo-code before and
that might be quite confusing for people who are not familiar with scala
reflection.

### CODE REVIEW

File: src/reflect/scala/reflect/api/Types.scala [hunk 1]
```diff
@@ -188,10 +188,12 @@ trait Types { self: Universe =>
...
-     *    class D[T] { def m: T }
-     *    class C extends p.D[Int]
-     *    T.asSeenFrom(ThisType(C), D)  (where D is owner of m)
-     *      = Int
+     *    class D[T] { def m: T = ??? }
+     *    class C extends D[Int]
+     *    val D = typeOf[D[_]].typeSymbol.asClass
+     *    val C = typeOf[C].typeSymbol.asClass
+     *    val T = D.typeParams(0).asType.toType
+     *    T.asSeenFrom(ThisType(C), D)
```
Comment (xeno-by) [inline src/reflect/scala/reflect/api/Types.scala]: What's the result of this invocation?

**Comment by xeno-by** [comment]
lgtm

**Comment by xeno-by** [comment]
@jsuereth This is just a documentation change, so it's safe to go into 2.10.0-RC2

**Comment by heathermiller** [comment]
I actually removed this example in the PR that I'm in the process of submitting.

**Comment by jsuereth** [comment]
I'm closing this, assuming Heather's fix was good enough.

Thanks for the request though!

**Comment by xeno-by** [comment]
@densh Please rebase and resubmit

**Pull request closed without merging**